New faceting syntax (#147)
posit-dev/ggsql: doc/syntax/clause/facet.qmd

## Clause syntax
The `FACET` syntax contains a number of subclauses that are all optional

```ggsql
FACET <column> BY <column>
    SETTING <parameter> => <value>, ...
    RENAMING <break-value> => <string>, ...
```

The first `column` is mandatory. It names a column in the layer data that will be used for splitting the data. If the layer data does not contain the column the behavior for that layer depends on the `missing` parameter of the facet.

### `BY`
The optional `BY` clause is used to define an additional column to split the data by. If it is missing the small multiples are laid out in a grid with the facet panels filling the cells in a row-wise fashion. If `BY` is present then the categories of the first `column` defines the rows of the grid and the categories of the second `column` the columns of the grid. Each multiple is then positioned according to that.

### `SETTING`
This clause behaves much like the `SETTINGS` clause in `DRAW`, in that it allows you to fine-tune specific behavior of the faceting. The following parameters exist:

> * `scales`: Controls whether the positional scales are kept fixed between the small multiples. Permissible values are `'fixed'` (default), `'free'`, `'free_x'`, and `'free_y'`. `'free'` allows both positional scales to vary between the panels, while the `free_x`/`free_y` versions keeps their counterpart fixed.

Author: Yeah, I kind of agree (with regard to naming)

> * `missing`: Determines how layers with a missing faceting column behaves. It can take two values: `'repeat'` (default), and `'null'`. If `'repeat'` is set then the layer data is repeated in each panel. If `'null'` then the faceting variable is set to `null` and only displayed if a null panel is shown (controlled by the facet scale).

Reviewer: ```suggestion
* `missing`: Determines how layers behave when the faceting column is missing. It can take two values: `'repeat'` (default), and `'null'`. If `'repeat'` is set, then the layer data is repeated in each panel. If `'null'`, then such layers are only displayed if a null panel is shown, as controlled by the facet scale.
```

* `ncol`: The number of columns to use when faceting by a single variable. Default is 3 when fewer than 6 categories are present, 4 when fewer than 12 categeries are present and otherwise 5

### `RENAMING`
This clause works much like the `LABEL` clause but works on the facet categories of the plot. The general syntax is that you provide the name of the category on the left and what it should appear as on the right, e.g `'adelie' => 'Pygoscelis adeliae'`. The clause is understood as a look-up table in the sense that if you provide a renaming for a category that doesn't appear in the facet then nothing will happen and if a category exist but doesn't have a renaming defined it will go through unaltered. To suppress the label of a specific category you can rename it to `null`, e.g. `'adelie' => null`. This will not remove the facet panel, only the label.

> #### Break formatting

Author: I have pinned for that as well
